paper/src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/SaplingHack.java

```diff
@@ -0,0 +1,95 @@
+package com.programmerdan.minecraft.simpleadminhacks.hacks;
+
+import com.programmerdan.minecraft.simpleadminhacks.SimpleAdminHacks;
+import com.programmerdan.minecraft.simpleadminhacks.configs.SaplingConfig;
+import com.programmerdan.minecraft.simpleadminhacks.framework.SimpleHack;
+import java.util.Random;
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.configuration.ConfigurationSection;
+import org.bukkit.enchantments.Enchantment;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.BlockBreakEvent;
+import org.bukkit.event.block.LeavesDecayEvent;
+import org.bukkit.inventory.ItemStack;
+
+public class SaplingHack extends SimpleHack<SaplingConfig> implements Listener {
+
+	private Random random = new Random();
+
+	public SaplingHack(SimpleAdminHacks plugin, SaplingConfig config) {
+		super(plugin, config);
+	}
+
+	@Override
+	public void onEnable() {
+		super.onEnable();
+		this.plugin.registerListener(this);
+	}
+
+	@EventHandler
+	public void onBlockBreak(BlockBreakEvent event) {
+		Block brokenBlock = event.getBlock();
+		if (!checkIfLeafBlock(brokenBlock)) {
+			return;
+		}
+		if (!this.config.isAllowFortune()) {
+			if (checkIfItemHasFortune(event.getPlayer().getItemInUse())) {
+				return;
+			}
+		}
+		if (!this.config.getChanceMap().containsKey(brokenBlock.getType())) {
+			return;
+		}
+		double nextDouble = random.nextDouble();
+		if (nextDouble <= this.config.getChanceMap().get(brokenBlock.getType())) {
+			brokenBlock.getWorld().dropItemNaturally(brokenBlock.getLocation(), new ItemStack(getSaplingMaterial(brokenBlock.getType()), 1));
+			logger.info("Dropped extra " + getSaplingMaterial(brokenBlock.getType())  + " from being broken at " + brokenBlock.getLocation());
+		}
+	}
+
+	@EventHandler
+	public void onLeafDecay(LeavesDecayEvent event) {
+		Block decayedBlock = event.getBlock();
+		if (!checkIfLeafBlock(decayedBlock)) {
+			return;
+		}
+		double nextDouble = random.nextDouble();
+		if (nextDouble <= this.config.getChanceMap().get(decayedBlock.getType())) {
+			decayedBlock.getWorld().dropItemNaturally(decayedBlock.getLocation(), new ItemStack(getSaplingMaterial(decayedBlock.getType()), 1));
+			logger.info("Dropped extra " + getSaplingMaterial(decayedBlock.getType())  + " from being broken at " + decayedBlock.getLocation());
+		}
+	}
+
+	private boolean checkIfLeafBlock(Block block) {
+		return switch (block.getType()) {
+			case OAK_LEAVES, BIRCH_LEAVES, SPRUCE_LEAVES, JUNGLE_LEAVES, ACACIA_LEAVES, DARK_OAK_LEAVES -> true;
+			default -> false;
+		};
```

Comment:
I'd replace this with `return getSaplingMaterial(block.getType()) != Material.AIR;` otherwise you're having to define the supported leaf types twice.

...or you could replace the entire method with a [Tag.LEAVES](https://jd.papermc.io/paper/1.18/org/bukkit/Tag.html#LEAVES=) check :P